modules/magniteAnalyticsAdapter.js

```diff
@@ -749,7 +764,8 @@ const handleBidResponse = (args, bidStatus) => {
 
   // if pbs gave us back a bidId, we need to use it and update our bidId to PBA
   const pbsBidId = (args.pbsBidId == 0 ? generateUUID() : args.pbsBidId) || (args.seatBidId == 0 ? generateUUID() : args.seatBidId);
-  if (pbsBidId) {
+  if (pbsBidId && !cache.bidsCachedClientSide.has(args)) {
+    cache.bidsCachedClientSide.delete(args);
```

Comment:
this works, but also defeats the purpose of the weakset :)
if you clean up, it will break when called multiple times; using a weak ref instead solves that, but it will break if the object identity changes.

This way you're vulnerable to both, but neither looks that likely to me so it's mostly an academic discussion -  LGTM any way you choose.

Reply:
This is why I value your input!

Will message you on slack to talk through it.

Reply:
You learn something new anytime you talk to Demetrio - thanks sir.